src/components/QueryForm.tsx

```diff
@@ -1,179 +1,97 @@
-        ) {
-          const filteredNode: FieldInputOption[] = matchedOptions.filter(
-            (n) => n.id !== 'allNodes'
-          );
-          setNode(filteredNode);
-          setSearchParams({ node: filteredNode.map((n) => n.label) });
-        } else {
-          setNode(matchedOptions);
-        }
-      }
-    }
-  }, [searchParams, setSearchParams, nodeOptions]);
-
-  function updateCategoricalQueryParams(fieldLabel: string, value: FieldInput) {
-    switch (fieldLabel) {
-      case 'Neurobagel graph':
-        setNode(value);
-        if (Array.isArray(value)) {
-          setSearchParams({ node: value.map((n) => n.label) });
-        }
-        break;
-      case 'Sex':
-        setSex(value);
-        break;
-      case 'Diagnosis':
-        setDiagnosis(value);
-        break;
-      case 'Assessment tool':
-        setAssessmentTool(value);
-        break;
-      case 'Imaging modality':
-        setImagingModality(value);
-        break;
-      default:
-        break;
-    }
-  }
+  const minAgeBiggerThanMax: boolean = minAge && maxAge ? minAge > maxAge : false;
+  const disableSubmit: boolean =
+    minAgeBiggerThanMax || minAgeError || maxAgeError || minNumSessionsError;
 
-  function updateContinuousQueryParams(fieldLabel: string, value: string | null) {
-    switch (fieldLabel) {
-      case 'Min age':
-        setMinAge(value);
-        break;
-      case 'Max age':
-        setMaxAge(value);
-        break;
-      case 'Minimum number of sessions':
-        setMinNumSessions(value);
-        break;
-      default:
-        break;
+  function validate(
+    label: string,
+    value: string,
+    setError: React.Dispatch<React.SetStateAction<boolean>>,
+    setHelperText: React.Dispatch<React.SetStateAction<string>>,
+    min: number,
+    max?: number
+  ) {
+    const numberValue: number = parseFloat(value);
+    if (Number.isNaN(numberValue) && value !== '') {
+      setError(true);
+      setHelperText('Value must be a number');
+      updateContinuousQueryParams(label, null);
+      return;
     }
-  }
-
-  /**
-   * Sets the value of a query parameter on the query parameter object.
-   *
-   * @remarks
-   * This is a utility function to used to help construct the query URL using a URLSearchParams object.
-   *
-   * @param param - The name of the query parameter
-   * @param value - The value of the query parameter
-   * @param queryParamObject - The query parameter object which contains the query parameters
-   * @returns void
-   */
-  function setQueryParam(param: string, value: FieldInput, queryParamObject: URLSearchParams) {
-    if (Array.isArray(value)) {
-      value.forEach((v) => {
-        queryParamObject.append(param, v.id);
-      });
+    if (numberValue < min) {
+      setError(true);
+      setHelperText(`Value must be greater than or equal to ${min}`);
+    } else if (max && numberValue > max) {
+      setError(true);
+      setHelperText(`Value must be less than or equal to ${max}`);
```

Comment:
as mentioned before, I think this whole function can go and the stateful error handling replaced by plain JS. This will also help with the slightly funky React.Dispatch type hints that are needed here